### ISSUE TITLE
Including `:pid` in `Sentry.LoggerBackend` metadata configuration silently makes errors not be sent to Sentry

### ISSUE THREAD

**Issue by kmteras** [getsentry/sentry-elixir#520]
### Environment

8.0.6

### Steps to Reproduce

- Had logger backend with the configuration
```
config :logger, Sentry.LoggerBackend,
  level: :error,
  metadata: [:pid],
  capture_log_messages: true

config :logger, :console,
  format: "$time $metadata[$level] $message\n",
  metadata: [:pid],
```

- Had an error be logged.

### Expected Result

Logged error is sent to Sentry.

### Actual Result

Logged error is not sent to Sentry and there is no indication to why.

Had to debug for a while why our logged errors are not being sent to Sentry. In the end it turned out it was having `:pid` be included for metadata. This was included in the whitelist for the logger and made sending errors to Sentry silently fail.

**Comment by whatyouhide** [comment]
The fix here is to `inspect/1` all metadata that doesn't implement the `String.Chars` protocol.

**Comment by whatyouhide** [comment]
This was closed with #602.
